### DISCUSSION TITLE
Contributing

### DISCUSSION THREAD

**Discussion by kennedybaird** [keystonejs/keystone#9244] [Questions]
I've been a long time fan of Keystone, and have used it in multiple different applications.

As has been noted by others, it seems on the surface that Keystone is quiet. The blog is coming close to 2 years since an update, the roadmap doesn't clearly indicate or represent some of the progress in recent history.

I have a little more spare time for the coming 6-8 weeks, and I would really like to help out - but it isn't clear where best I offer energy.

There are some (larger) areas where I feel like I would love to look at contributing:
- roles, improving access control
- list extensions / plugin framework?
  - a simple example in a previous project was adding an "event" extension, which added `afterOperation` hooks for adding an `AdminEvent` for fields
- context in custom admin pages
However, there are some areas that may seem more pressing:
- updating nextjs to use the new app router
- nested fields
- sortable lists
- helping with GH issues / discussions

Any thoughts would be wonderful, I want to be conscious of not adding more to the current maintainer's plates as well - and would welcome being contacted privately if that suits :pray:

**Comment by gautamsi** [comment]
One of the area I think is converting design system and list pages to server components with help of App Router. #9183.

My idea is to have things ready with React 19 and Nextjs 15 when they finally release. I am new to this and have been mostly working to get this converted to App router with minimal changes so far. #9186

**Reply by kennedybaird** [reply]
Thanks - what in particular would you like on #9186?

**Reply by dcousens** [reply]
@gautamsi @kennedybaird stay tuned for a draft pull request from @jossmac for a design system upgrade.

@kennedybaird, testing and development would be immeasurably appreciated as many fields will need upgrading to the new design system. Probably 2 weeks away from a skeleton draft.

See the https://github.com/keystonejs/keystone/tree/introduce-keystar-ui branch if you want to jump in early

**Reply by dcousens** [reply]
I put a draft pull request up https://github.com/keystonejs/keystone/pull/9253 so we can track a discussion and relevant changes.

**Reply by kennedybaird** [reply]
Thanks @dcousens - are you looking for general UI feedback in there at this stage as well?

**Reply by dcousens** [reply]
@kennedybaird I think we're in draft, nearly any feedback is welcome if constructive; nits and fixes would probably be better appreciated as patches though :yellow_heart:

**Reply by arihantdaga** [reply]
Hello all! Is there some documentation on setting up development server for keystone ?
How can i test changes made in keystone packages directly in my example app, without having to manually rebuild keystone everytime i change something.
